File: src/core/server/Database/Models/Character.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Numerics;
using triallife.Helper;

namespace triallife.Database.Models {
	public class Character {
		[Key]
		public int id { get; set; }
		public int accId { get; set; }
		public Vector3 pos { get; set; } = new Vector3(-867.1437377929688f, -172.6201934814453f, 37.799232482910156f);
		public string name { get; set; }
		public int bankId { get; set; } = 0;
		public int armour { get; set; } = 0;
		public double money { get; set; } = 500;
		public double blood { get; set; } = 7500;
		public double hunger { get; set; } = 100;
		public double thirst { get; set; } = 100;
		public double mood { get; set; } = 100;
		public bool isUnconsciouse { get; set; } = false;
		public int hours { get; set; } = 0;
		public string interior { get; set; }
		public Vector3 exterior { get; set; }
		public Design design { get; set; } = new Design();
		public CharInfo info { get; set; } = new CharInfo();
		public Inventory inventory { get; set; } = new Inventory() { maxWeight = 15, items = new List<Item>() };
		public List<Item> equipment { get; set; } = new List<Item>(8);
		public List<Item> toolbar { get; set; } = new List<Item>(5);
		public List<Vehicle> vehicles { get; set; } = new List<Vehicle>();
	}

	public class CharInfo {
		public string gender { get; set; }
		public int age { get; set; }
	}
}

File: src/core/shared/interfaces/character.ts
import { Vector3 } from 'alt-server';
import { DefaultConfig } from '../../server/configs/settings';
import { Bank } from './bank';
import { Design } from './design';
import { Item } from './item';
import { Vehicle } from './vehicle';

export interface Character {
    _id?: any;
    accId: any;
    pos: Partial<Vector3>;
    bank: Partial<Bank>;
    info: Partial<CharacterInfo>;
    stats: Partial<CharacterStats>;
    isUnconsciouse: boolean;
    hours: number;
    interior: string | null;
    exterior: Partial<Vector3>;
    design: Partial<Design>;
    inventory: { maxWeight: number; money: number; items: Array<Partial<Item>> };
    equipment: Array<Partial<Item>>;
    toolbar: Array<Partial<Item>>;
    vehicles: Array<Partial<Vehicle>>;
}

export const CharacterDefaults: Partial<Character> = {
    pos: DefaultConfig.PLAYER_CREATE_SPAWN_POS as Vector3,
    inventory: { maxWeight: 15.0, money: DefaultConfig.PLAYER_CASH, items: [] },
    bank: {},
    design: {},
    info: {},
    stats: { blood: 7500, hunger: 100, thirst: 100, mood: 100, armour: 0 },
    isUnconsciouse: false,
    hours: 0,
    vehicles: []
};

export interface CharacterInfo {
    gender: string;
    name: string;
    age: number;
}

export interface CharacterStats {
    blood: number;
    hunger: number;
    thirst: number;
    mood: number;
    armour: number;
}
